keras/utils/generic_utils.py

```diff
@@ -461,6 +461,22 @@ def to_list(x):
     return [x]
 
 
+def first_or_list(x):
+    """ Get the first element if the iterable has only one value.
+
+    Otherwise return the iterable
+
+    # Argument:
+        x: An iterable
```

Comment:
List or tuple. Iterables implement `__iter__()` and thus may not necessarily be indexable (requires `__getitem__()`).